girder/models/user.py

```diff
@@ -244,26 +249,43 @@ def createUser(self, login, password, firstName, lastName, email,
             'groupInvites': []
         }
 
-        self.setPassword(user, password)
+        self.setPassword(user, password, save=False)
+
+        self.setPublic(user, public, save=False)
+
+        # An id is required in order to grant the user access on itself, but
+        # since the user isn't fully created yet, it should not be saved
+        user['_id'] = ObjectId()
```

Comment:
What I really care about is that callbacks listening for ``model.{}.save.created`` receive a completely created object, with no extra creation work still pending to execute.

Note, what originally bit me is that I had a callback to set all newly-created users to be private. However, since the ``model.user.save.created`` was first triggered inside ``setPassword``, the work of my callback function was always undone.

The notion of atomic creation is nice, but not as practically important IMO.